sklearn/inspection/tests/test_partial_dependence.py

```diff
@@ -340,16 +345,32 @@ def test_partial_dependence_error(estimator, params, err_msg):
     'estimator',
     [LinearRegression(), GradientBoostingClassifier(random_state=0)]
 )
-@pytest.mark.parametrize('features', [-1, 1000000])
-def test_partial_dependence_unknown_feature(estimator, features):
+def test_partial_dependence_unknown_feature_indices(estimator):
```

Comment:
I'm not convinced that we want to support negative indexes for features in a user-facing API, but not strong opinion

At least worth a new test case in `test_partial_dependence_feature_type`?

Reply:
That's why we check that we raise an error for negative indices.

Reply:
Where? It looks like you removed the parametrization

Reply:
Oh you are right.

Reply:
So after checking a bit more, we are supporting negative in the `ColumnTransformer`. However, the error message mentioned to have feature indices between [0, n_features], so not really sure that we wanted to support that in the first place.

Reply:
I don't know about the ColumnTransformer but let's just keep things simple here and not support it. If people want it, we can add it later.

Reply:
Uhm this is actually a feature because we have a test for it.

Reply:
Sorry, I don't undersand.

The current state of the master branch is to not support negative indexing. There's a test that makes sure we don't. We should keep it this way.

Reply:
ping @glemaitre can you please revert support for negative indexing here?

Reply:
`ColumnTransformer` is supporting negative indexing. We have a test for the `ColumnTransformer`.
Thus, if someone writes a valid pipeline having negative indexing in a `ColumnTransformer`, there is no reason to raise an error within the `partial_depedence` computation. It should just work.

Reply:
I'm sorry but I really don't understand what ColumnTransformer has to do in the discussion here. We are considering the API of `partial_dependence` and `plot_partial_dependence`.

Can you please give a code example where not supporting negative indexing is an issue?

Reply:
Oh, sorry you are right. Getting a bit confused between my PR sorry.

Here the only unpleasant thing will be that you can create a column transformer with negative indexing and to compute the PD of this feature you will need to provide the positive counterpart. As you mentioned, we can let this happened if people request it.

Reply:
```python
proc = make_column_transformer(('trans', FunctionTransformer, [-1, -2]))
pipe = make_pipeline(proc, LogisticRegression())
pipe.fit(X, y)
# Now you need to call the plot_partial_dependence with positive.
plot_partial_dependence(pipe, X, feature=[X.shape[1], X.shape[1] - 1)
# This would be more consistent
plot_partial_dependence(pipe, X, feature=[-1, -2])
```